resources/inertia/Components/Transaction/TransactionsTable.tsx

```diff
@@ -0,0 +1,77 @@
+import { TransactionsProps } from "@/Pages/Transactions.contracts";
+import FilterProvider from "@/Providers/Filter/FilterProvider";
+import { useTranslation } from "react-i18next";
+import TransactionsTableWrapper from "@/Components/Tables/Desktop/Transactions/Transactions";
+import TransactionsMobileTableWrapper from "@/Components/Tables/Mobile/Transactions/Transactions";
+import { router } from "@inertiajs/react";
+import { usePageHandler } from "@/Providers/PageHandler/PageHandlerContext";
+import { useEffect } from "react";
+
+export default function TransactionsTable({
+    transactions,
+    filters,
+}: Pick<TransactionsProps, "transactions" | "filters">) {
+    const { t } = useTranslation();
+    const { setRefreshPage } = usePageHandler();
+
+    useEffect(() => {
+        setRefreshPage(() => {
+            router.reload({
+                only: ["transactions", "filters"],
+            });
+        });
+    }, []);
```

Comment:
I had the same issue - pagination gets stuck in the loading state. You need to do something like this:

```
    const { setRefreshPage } = usePageHandler();

    const updateTable = (callback?: CallableFunction) => {
        router.reload({
            only: ["blocks"],
            onSuccess: () => {
                if (callback) {
                    callback();
                }
            },
        });
    };

    useEffect(() => {
        updateTable();

        setRefreshPage((callback: CallableFunction) => {
            updateTable(callback);
        });
    }, []);
```